Framework/Beamline/inc/MantidBeamline/DetectorInfo.h

```diff
@@ -57,13 +57,17 @@ namespace Beamline {
 class MANTID_BEAMLINE_DLL DetectorInfo {
 public:
   DetectorInfo(const size_t numberOfDetectors);
+  DetectorInfo(const size_t numberOfDetectors,
+               const std::vector<size_t> &monitorIndices);
 
   size_t size() const;
 
+  bool isMonitor(const size_t index) const;
   bool isMasked(const size_t index) const;
   void setMasked(const size_t index, bool masked);
```

Comment:
Do we also need to consider eventually `setMonitor` or would you do that at construction time? You could decide to use a detector or group of detectors as a monitor.